### PR TITLE
feat: added support to restore DB from backup id

### PR DESCRIPTION
**Pull request by Ak-sky**
### Description

Added support to restore DB from backup id.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Added support to restore DB from backup id.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: examples/backup/main.tf [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  rabbitmq_version  = var.rabbitmq_version
+}
+
+data "ibm_database_backups" "backup_database" {
+  count         = var.rmq_db_backup_crn != null ? 0 : 1
+  deployment_id = module.rmq_db[0].id
+}
+
+# New RabbitMQ db instance pointing to the backup instance
+module "restored_rmq_db" {
```
Comment (shemau) [inline examples/backup/main.tf:34]: Can we avoid the use of rqm and rqm_db and prefer the use of rabbitmq or possibly rabbitmq_database (per fscloud module name).
Follow the other examples, the modules could be icd_rabbitmq (or rabbitmq_database per fscloud module) and restored_icd_rabbitmq.

File: examples/backup/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+output "restored_rmq_db_id" {
+  description = "Restored RabbitMQ db instance id"
+  value       = module.restored_rmq_db.id
+}
+
+output "restored_rmq_db_version" {
+  description = "Restored RabbitMQ instance version"
+  value       = module.restored_rmq_db.version
+}
```
Comment (shemau) [inline examples/backup/outputs.tf:18]: Can we avoid the use of rqm and rqm_db and prefer the use of rabbitmq or possibly rabbitmq_database (per fscloud module name).
Maybe restored_rabbitmq_id and restored_rabbitmq_version

File: examples/backup/variables.tf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the rabbitmq instance created by the module, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial for more details"
+  default     = []
+}
+
+variable "rmq_db_backup_crn" {
+  type        = string
```
Comment (shemau) [inline examples/backup/variables.tf:44]: Can we avoid the use of rqm and rqm_db and prefer the use of rabbitmq or possibly rabbitmq_database (per fscloud module name).
Follow the example of rabbitmq_version, the variable could be rabbitmq_backup_crn.

**Review by shemau — CHANGES_REQUESTED** [review]
Generally looks good. rqm has not been used previously in this module, so I wonder if sticking with rabbitmq is more consistent.

**Comment by Ak-sky** [comment]
- Able to restore mongodb instance from the backup id-

   image

- Restored instance details- 

   image

File: examples/backup/main.tf (hunk 1 above)
Comment (Ak-sky) [reply examples/backup/main.tf]: Will add the example later and will use rabbitmq.

File: examples/backup/variables.tf (hunk 3 above)
Comment (Ak-sky) [reply examples/backup/variables.tf]: Will add the example later and will use rabbitmq

File: examples/backup/outputs.tf (hunk 2 above)
Comment (Ak-sky) [reply examples/backup/outputs.tf]: Will add the example later and will use rabbitmq

**Review by ocofaigh — APPROVED** [review]

**Review by shemau — APPROVED** [review]
LGTM

**Pull request merged**